### DISCUSSION TITLE
Add support for s3 driver to social-image generation

### DISCUSSION THREAD

**Discussion by SanderGo** [studio1902/statamic-peak-seo#16] [Ideas]
Hi Rob,

Today I ran into an issue where I couldn't generate the social-images. Finally figured out my filesystem config for the social_images disk is causing the issue. I'm using DigitalOcean Spaces with the s3 driver. 

For the other disks it works just fine, but for Browsershot the save method needs a path to save to. 

I've managed to fix the problem by adding a driver check. When using the s3 driver, it puts the screenshot to the disk. Or else it will perform the save method just like before.

GenerateSocialImagesJob.php:73-77

```
$image = Browsershot::url("{$absolute_url}/social-images/{$id}")
            ->windowSize(1200, 630)
            ->select('#og')
            ->waitUntilNetworkIdle()
            ->save($disk->path($file));
```

to

```
$image = Browsershot::url("{$absolute_url}/social-images/{$id}")
            ->windowSize(1200, 630)
            ->select('#og')
            ->waitUntilNetworkIdle();

if (strtolower(config("filesystems.disks.{$container->disk}.driver") == 's3')) {
    $disk->put($file, $image->screenshot());
} else {
    $image->save($disk->path($file));
}
```

Works fine for me on both DO and with the local driver. If you agree I could PR the update. 

Greets, Sander

**Comment by robdekort** [comment]
That sounds like a great solution to support multiple use cases. I'd gladly accept a PR. Thanks.

**Reply by SanderGo** [reply]
Cool! I'll submit the PR. Thanks!

**Reply by robdekort** [reply]
Thank you!

**Reply by SanderGo** [reply]
I've submitted the PR. It's my first public PR, so any feedback is welcome :) Tried to follow the 'rules' and guidelines as good as I could.

**Reply by robdekort** [reply]
Looks great Sander. No worries, I'll dive in. Thanks.
